src/Avalonia.Controls/Primitives/Popup.cs

```diff
@@ -637,7 +637,7 @@ private void PassThroughEvent(PointerPressedEventArgs e)
             if (e.Source is LightDismissOverlayLayer layer &&
                 layer.GetVisualRoot() is IInputElement root)
             {
-                var p = e.GetCurrentPoint(root);
+                var p = e.GetCurrentPoint(layer.GetVisualRoot());
```

Comment:
If feels wrong because it reads visual root twice here

Reply:
I can split the if and just do it in more code to avoid the second lookup.

Reply:
Alright, I've changed it to only query it once.